poly-commitment/src/srs.rs

```diff
@@ -27,10 +27,10 @@ pub struct SRS<G: CommitmentCurve> {
     pub lagrange_bases: HashMap<usize, Vec<G>>,
     /// Coefficient for the curve endomorphism
     #[serde(skip)]
-    pub endo_r: G::ScalarField,
+    pub(crate) endo_r: G::ScalarField,
     /// Coefficient for the curve endomorphism
     #[serde(skip)]
-    pub endo_q: G::BaseField,
+    pub(crate) endo_q: G::BaseField,
```

Comment:
I think you'll have some issues with the ocaml side with that change, as they try to set these fields directly. (That being said, I think we now have a way to not set these fields directly from there thanks to @chee-chyuan 's changes.)

Reply:
also why can't we just delete these fields from the SRS now?

Reply:
I thought that too, it seems to use the endos in for `ScalarChallenge::to_field()`, the endo_q is used only once but passed to a function that ignores it, so only the endo_r should be necessary currently.

Reply:
maybe this is something that can be manually passed to the function that needs it?

Reply:
btw I know this is a lot of changes for this PR :D if you want we can keep things this way and try to clean up the SRS in another PR?

Reply:
Sounds good, for now I will just let them as pub and we can revisit it in the future in a more general refactor of `/poly-commitment`.